### PR TITLE
RFP FILExplorer AHC Team

### PR DESCRIPTION
**Pull request by lrgeoemtry**
Update to Proposal as of Aug 30, 2019

*Subject to change, expand, evolve*

### CODE REVIEW

File: rfp-proposals/rfp-proposal-filexplorer-AHC.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+Hi there, we're the AHC Development team. We've been enjoying IPFS for sometime and would like to be a part of this jounrey as it scales. We plan to have our designer whip up some really sleek UI for a Block Explorer we would use (as devs and a use case) as well as one that others interested in the FIL ecosystem would enjoy and understand intuitively.
+- For Milestone 1 we will have the mocks and the basic ui on a staging environment, and hopefully if the FIL docs are good we'll have basic realtime dev data from the FIL network populating the fields. 
+- For Milestone 2 we'd like to work closely with those in charge to ensure Milestone 1 is behind us while improving the useability and complete functionality of the explorer. 
+- For Milestone 3 we'll have everything documented appropiately so anyone can pick up and build it while deciding an appropiate maintainence agreement post build. 
+
+
+## Deliverables
+
+- Fully functional block explorer website, hosted/maintained/managed by your team. Publicly accessible via a `https://filexplorer.org` site.
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-filexplorer-AHC.md]: By "your team", do you mean the Protocol Labs-Filecoin team? We are looking for teams that will maintain their own projects, so would be helpful to clarify!

File: rfp-proposals/rfp-proposal-filexplorer-AHC.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  - Amazon EC2 Hosted Instance for Cloud Compute & CloudFlare for DDOS Protection
+  - Good UX and visual design for both Mobile & Desktop 
+  - Real-time updates based on changes in the network (dev / production)
+  - Fast to load across any region
+  - Open-sourced repo for block explorer codebase, dual-licensed under MIT & Apache 2.0
+  - Well-documented codebase so anyone can learn from the application development process (ReadtheDocs)
+  - User-facing documentation for anyone to spin up the fil explorer locally
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-filexplorer-AHC.md]: Great!

File: rfp-proposals/rfp-proposal-filexplorer-AHC.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+
+## Detailed Requirements & Constraints
+
+### User Flows `Brought in For Team's Convienence`
+This tool will most often be used by Filecoin developers, token holders, clients, and miners to investigate transactions, debug their nodes, or generally explore the chain. We have done some thinking about the user needs we would like this tool to satisfy. You can use this as a starting point or take this in a totally new direction. Some example user stories:
+
+- *Prospective Client*: I am considering using Filecoin for my data storage needs. I decide to visit the Filecoin block explorer to get a pulse on important storage contract parameters before using the network.
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-filexplorer-AHC.md]: Do you have examples of the sorts of storage contract parameters you're thinking of?

File: rfp-proposals/rfp-proposal-filexplorer-AHC.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+- *Current Client*: I recently hired a Filecoin storage miner to store some of my data. I want to use the Filecoin block explorer to easily monitor the status of my storage providers, keeping tabs on their deal activity and any potential reputational metrics.
+- *Potential Miner*: Before signing up and pledging collateral to mine Filecoin, I want to see how frequently blocks are mined, the size of the block reward, and storage deal information so I can understand mining in more detail.
+- *Current Storage Miner*: I haven’t mined a block in a while and I want to check the chain state to make sure that my node is in sync with the rest of the chain, and to make sure that the network is still working as it should. I want to be able to see the nodes coming to consensus and understand mining statistics such as when the last block was mined.
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-filexplorer-AHC.md]: Is this the main user story for current storage miners?

File: rfp-proposals/rfp-proposal-filexplorer-AHC.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+- *Developer*: I am building an application on top of Filecoin and want to ensure that my application is in sync with the latest chain-state information. I also want to confirm my deal information with corresponding block data using the block explorer.
+- *Accountants / Investors*: I want to be able to retrieve a list of transaction details for a period of time as well as ending token balances per wallet for tax and accounting purposes.
+
+### Data to Display
+We anticipate being able to provide a homepage showing summary statistics, with the ability to click for more detailed information as requested in the RFP.
+
+- **Top Level / Home Page Summary Statistics**: *Clean, streamlined view of the network. Ideally most people will get all the top level stats they’re looking for from this page without having to navigate away from it.*
+	- [Creative visualization](http://www.ethviewer.live/) of blocks mined and nodes (or peers) coming to consensus / [visualization of the blockchain growing in real time](https://user.kittyhawk.wtf:8000)
+	- Include a [universal search bar](https://blockstream.info/) that allows users to search by block height, transaction, block hash, or address
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-filexplorer-AHC.md]: What about searching by data CID?

File: rfp-proposals/rfp-proposal-filexplorer-AHC.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+	- Prioritized Block Stats including some, perhaps not all, of the following:
+		- Tipset Height
+		- Number of blocks in tipset: Number of blocks in the current best tipset. A tipset is a set of blocks at the same height that share the same parent set.
+		- Hash of each parent block in a tipset
+		- Time since last block: Time since a miner last created a block
+		- Avg. block time: Average time between blocks that are created / mined
+		- Last block miner address
+		- Last block peerID
+		- Storage Power (%): The probability that a storage miner will win the ability to mine the next block. Miners gain power by storing data, sealing it, and proving they still have it (i.e., providing a Proof-of-Spacetime).
+		- Ticket
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-filexplorer-AHC.md]: Why would this be useful to prioritize?

File: rfp-proposals/rfp-proposal-filexplorer-AHC.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+- **Secondary Views: Detailed Blockchain Metrics**: *Each bullet is a separate tab or view, though it may make sense to combine some. Tabs: Explorer (home page above), Block Data, Deals, Actors*
+	- High Level Deal Data:
+		- Average deal price (normalized to FIL/GB/month or $/GB/month) over time
+		- Interactive graph of deal volume over time
+		- Average deal duration 
+		- Average deal size
+		- Average deal replication factor
+		- Average sector size
+		- Average # of deals / block
+	- Transaction Detail Table:
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-filexplorer-AHC.md]: How would you get this data?

**Comment by AndreiD** [comment]
just a comment, Developer= $250/hour.... -> According to PayScale, the average hourly rate of the software developer in USA is $34,65. AS their expertise gets higher, the rate can climb up to $62,14.... now. that must be one hell of a developer to charge $250 per hour :)

**Comment by pooja** [comment]
We've finished the wave 1 grant selection process. Can you please email devgrants@filecoin.org so we can share our decision and next steps, if any? In your email, please include a link to your proposal PR so we know who we're talking to!

Thanks,
Filecoin DevGrants team

**Comment by pooja** [comment]
Greetings,

Thank you for your application to the Filecoin DevGrants program. We appreciate your proposal!

You were not selected for a grant in Wave 1, but we hope that you consider applying again for a future DevGrant in a later wave.

We selected grants based on:
- what makes sense to build now based on where the Filecoin node implementations are
- the likelihood of your team being able to work independently of the Filecoin engineering teams working on the core client implementations
- degree of refactoring your proposal may require as the software clients evolve
- the integrity of the proposal and near-term value to the Filecoin ecosystem
- your commitment to the ecosystem and desire to maintain your project longer-term

As the Filecoin implementations advance towards Testnet and Mainnet, we hope you continue to consider new proposals for projects. We would love to see more applications from you in the future.

Please stay tuned for more Filecoin grant RFPs. :)

**Comment by lrgeoemtry** [comment]
@pooja argh just saw allllll these comments, no idea why I didnt get the notifcations. 
Will go in for dev grant 2 and refactor to fit the needs of the community

**Pull request closed without merging**